enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,134 @@
+---
+title: use-manifest-annotation-for-object-removal
+authors:
+  - "@jottofar"
+reviewers:
+  - "@LalatenduMohanty"
+  - "@wking"
+  - "@crawford"
+  - "@sdodson"
+approvers:
+  - "@crawford"
+  - "@sdodson"
+creation-date: 2020-08-03
+last-updated: 2020-08-04
+status: implementable
+---
+
+# Use Manifest Annotation For Object Removal
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently CVO managed object removal is handled by jobs.
+Jobs have many immutable components, so they aren't a great match for CVO managed manifests.
+This enhancement replaces that approach with a new manifest annotation requesting the CVO delete the in-cluster object instead of creating/reconciling it.
+This will provide a more straightforward way to remove content.
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
```

Comment:
... for at least N+1 versions of payload.

You should probably define the human and review process around using these.  How do we ensure they get cleaned up in N+1?

Reply:
Would bugzilla suffice? I assume a Jira (RFE) is created to do the original removal and as part of that a bug is created against N+1 to remove the manifest itself. Part of this enhancement is to update the CVO dev docs adding this delete feature and the former procedure could be laid out there.

Reply:
> I assume a Jira (RFE) is created to do the original removal and as part of that a bug is created against N+1 to remove the manifest itself.

Say you wanted to remove object A in 4.6.  Target Release 4.7 does not exist today.  There's not actually a problem leaving these around into N+2, etc., because we should only be removing stuff that's restricted to core-component control (so deletion-manifests won't squat on any names users should be using).  I'm fine without a formal procedure around dropping the deletion manifests, as long as we keep them for the whole z stream they were born in.

Reply:
The cleanup is the responsibility of the operator. Can CVO do anything about it? I mean even if we come up with a human process the operator devs need to follow it and CVO can not do anything if they do not follow it.

Reply:
@smarterclayton Do you agree with @wking and @LalatenduMohanty comments and if not where are we differing?